mars/dataframe/reduction/core.py

```diff
@@ -0,0 +1,305 @@
+# Copyright 1999-2018 Alibaba Group Holding Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import pandas as pd
+
+from ...config import options
+from ...serialize import BoolField, AnyField, DataTypeField, Int32Field
+from ..utils import parse_index, build_empty_df
+from ..operands import DataFrameOperandMixin, DataFrameOperand, ObjectType, SERIES_CHUNK_TYPE
+from ..merge import DataFrameConcat
+
+
+class DataFrameReductionOperand(DataFrameOperand):
+    _axis = AnyField('axis')
+    _skipna = BoolField('skipna')
+    _level = AnyField('level')
+    _min_count = Int32Field('min_count')
+    _need_count = BoolField('need_count')
+
+    _dtype = DataTypeField('dtype')
+    _combine_size = Int32Field('combine_size')
+
+    def __init__(self, axis=None, skipna=None, level=None, min_count=None, need_count=None, dtype=None,
+                 combine_size=None, gpu=None, sparse=None, **kw):
+        super(DataFrameReductionOperand, self).__init__(_axis=axis, _skipna=skipna, _level=level, _min_count=min_count,
+                                                        _need_count=need_count, _dtype=dtype,
+                                                        _combine_size=combine_size, _gpu=gpu, _sparse=sparse, **kw)
+
+    @property
+    def axis(self):
+        return self._axis
+
+    @property
+    def skipna(self):
+        return self._skipna
+
+    @property
+    def level(self):
+        return self._level
+
+    @property
+    def min_count(self):
+        return self._min_count
+
+    @property
+    def need_count(self):
+        return self._need_count
+
+    @property
+    def dtype(self):
+        return self._dtype
+
+    @property
+    def combine_size(self):
+        return self._combine_size
+
+
+class ReductionMixin(DataFrameOperandMixin):
+    @classmethod
+    def _get_reduced_meta(cls, chunk, axis):
```

Comment:
I think this function can be removed. It's signature is different according to different situation, it's hard to maintain, and I think it's unnecessary to calculate meta for each dataframe, series and chunks.